ext/storage_sources/s3_store/test/test_s3_connection.cpp

```diff
@@ -24,15 +27,207 @@ int TestListBuckets(const Aws::S3Crt::ClientConfiguration &config);
+            std::cout << "  * " << object << std::endl;
+
+        /* List with total_objects < max_per_iter. */
+        if (ret = conn.ListObjects(
+                    firstBucket, "test_list_objects_", objects, countp, maxPerIter, 4) != 0)
+            goto err;
+        if (objects.size() != countp || countp != 4) {
+            /* Delete objects and file at end of test. */
+            for (int i = 0; i < 20; i++) {
+                if (ret = conn.DeleteObject(
+                            firstBucket, "test_list_objects_" + std::to_string(i) + ".txt") != 0)
+                    goto err;
+            }
+            std::remove("test_list_objects.txt");
+            ret = 1;
+            goto err;
+        }
+        std::cout << "List with total objects < max per iter:" << std::endl;
+        for (const auto &object : objects)
+            std::cout << "  * " << object << std::endl;
+
+        /* List with total objects non-divisable by max per iter. */
+        if (ret = conn.ListObjects(
+                    firstBucket, "test_list_objects_", objects, countp, maxPerIter, 8) != 0)
+            goto err;
+        if (objects.size() != countp || countp != 8) {
+            /* Delete objects and file at end of test. */
+            for (int i = 0; i < 20; i++) {
+                if (ret = conn.DeleteObject(
+                            firstBucket, "test_list_objects_" + std::to_string(i) + ".txt") != 0)
+                    goto err;
+            }
+            std::remove("test_list_objects.txt");
+            ret = 1;
+            goto err;
+        }
+        std::cout << "List with total objects non-divisable by max per iter:" << std::endl;
+        for (const auto &object : objects)
+            std::cout << "  * " << object << std::endl;
+
+        /* List with max objects greater than total objects. */
+        if (ret = conn.ListObjects(firstBucket, "test_list_objects_", objects, countp, 5, 30) != 0)
+            goto err;
+        if (objects.size() != countp || countp != 20) {
+            /* Delete objects and file at end of test. */
+            for (int i = 0; i < 20; i++) {
+                if (ret = conn.DeleteObject(
+                            firstBucket, "test_list_objects_" + std::to_string(i) + ".txt") != 0)
+                    goto err;
+            }
+            std::remove("test_list_objects.txt");
+            ret = 1;
+            goto err;
+        }
+        std::cout << "List all:" << std::endl;
+        for (const auto &object : objects)
+            std::cout << "  * " << object << std::endl;
+
+        /* Delete objects and file at end of test. */
+        for (int i = 0; i < 20; i++) {
+            if (ret = conn.DeleteObject(
+                        firstBucket, "test_list_objects_" + std::to_string(i) + ".txt") != 0)
+                goto err;
+        }
+        std::remove("test_list_objects.txt");
+    } else {
+        std::cout << "No buckets in AWS account." << std::endl;
```

Comment:
Is it considered as a failure? Should we set `ret` to `1`?

Reply:
I have changed this to return `1` when no buckets are found in the AWS account. I don't think this is too important, as `ListBuckets` will be removed soon and we will be using a shared account, with bucket name as a class member variable of the S3 connection.